File: delete-operation-for-two-strings/delete-operation-for-two-strings.cpp
class Solution {
public:
    int minDistance(string word1, string word2) {
        int m=word1.size(),n=word2.size();
        int t[m+1][n+1];
        memset(t,0,sizeof(t));
        for(int i=1;i<=m;i++)
        {
            for(int j=1;j<=n;j++)
            {
                if(word1[i-1]==word2[j-1])
                    t[i][j]=t[i-1][j-1]+1;
                else
                    t[i][j]=max(t[i-1][j],t[i][j-1]);
            }
        }
        return m+n-2*t[m][n];
    }
};

File: 583-delete-operation-for-two-strings/delete-operation-for-two-strings.py
class Solution:
    def minDistance(self, s1: str, s2: str) -> int:
        n = len(s1)
        m = len(s2)
        dp = [[-1]*m for _ in range(n)]
        def solve(i,j):
            if i == n and j == m:
                return 0
            if i == n :
                return m - j
            if j == m :
                return n - i
            if dp[i][j] != -1:
                return dp[i][j]
            if s1[i] == s2[j]:
                dp[i][j] = solve(i+1,j+1)
            else:
                dp[i][j] =  1 + min(solve(i+1,j), solve(i, j+1))
            return dp[i][j]
        return solve(0,0)